src/main/java/UseCase/RegisterInteractor.java

```diff
@@ -0,0 +1,46 @@
+package UseCase;
+import Entities.User;
+import OutputBoundary.loginOutputBoundary;
+import OutputData.loginOutputData;
+import InputData.loginInputData;
+import InputBoundary.RegisterInputBoundary;
+import Database.userAccessInterface;
+import Database.userDsData;
+import Entities.userFactory;
+
+public class RegisterInteractor implements RegisterInputBoundary{
+    private final loginOutputBoundary registerPresenter;
+    private final userFactory userFactory;
+
+    userAccessInterface Allusers;
+
+    public RegisterInteractor(loginOutputBoundary registerPresenter, userAccessInterface Allusers, userFactory userFactory){
+        this.Allusers = Allusers;
+        this.registerPresenter =registerPresenter;
+        this.userFactory = userFactory;
+    }
+    @Override
+    public void register(loginInputData loginID){
+        if (Allusers.exists(loginID.getUsername())){
+            //TODO: prepare user exists failure view
+        } else if (loginID.getPassword() == "") {
+            //TODO: prepare no password failure view
+        }
+
+        if (loginID.isArtist()){
+            User user = this.userFactory.createArtistUser("", loginID.getUsername(), loginID.getPassword());
```

Comment:
Hold up -- why is artistName being initialized as ""? The variable within the ArtistUser entity is final anyway, so it's not capable of being changed at a later point.

You might need to make a separate ArtistInputData (or some other name) for registering artists, so we can get that name.

Reply:
the thing is that I thought we dont take in artist name at login? so do u mean like adding an extra input for artist registration?